### PR TITLE
kv: optimize IsPoint

### PR DESCRIPTION
**Pull request by coocood**
### What problem does this PR solve? 
Improve performance of `IsPoint`.
```
benchmark               old ns/op     new ns/op     delta
BenchmarkIsPoint-12     5.35          21.1          +294.39%

benchmark               old allocs     new allocs     delta
BenchmarkIsPoint-12     0              1              +Inf%

benchmark               old bytes     new bytes     delta
BenchmarkIsPoint-12     0             8             +Inf%
```

### What is changed and how it works?
Avoid memory allocation by comparing bytes directly.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: kv/key.go [hunk 1]
```diff
@@ -78,7 +78,25 @@ type KeyRange struct {
 
 // IsPoint checks if the key range represents a point.
 func (r *KeyRange) IsPoint() bool {
-	return bytes.Equal(r.StartKey.PrefixNext(), r.EndKey)
+	if len(r.StartKey) != len(r.EndKey) {
```
Comment (shenli) [inline kv/key.go]: Please add some comments in this function.

Comment (tiancaiamao) [reply kv/key.go]: The new code is very hacking, I think it's better add commit like this:

`// return bytes.Equal(r.StartKey.PrefixNext(), r.EndKey)`

File: kv/key.go [hunk 2]
```diff
@@ -78,7 +78,34 @@ type KeyRange struct {
...
+			break
+		}
+		if r.EndKey[i] != 0 {
+			return false
+		}
+		all255SuffixIdx = i
+	}
+	// The byte at diffIdx in StartKey should be one less than the byte at diffIdx in EndKey.
+	// And bytes in StartKey and EndKey before diffIdx should be equal.
+	diffIdx := all255SuffixIdx - 1
```
Comment (crazycs520) [inline kv/key.go]: how about:
```go
	i := len(r.StartKey) - 1
	for ; i >= 0; i-- {
		if r.StartKey[i] != 255 {
			break
		}
		if r.EndKey[i] != 0 {
			return false
		}
	}
	diffIdx := i
```
and rename `i` to `diffIdx` will be better.

**Comment by coocood** [comment]
@crazycs520 PTAL

**Pull request merged**